static/js/controllers/contacts.js

```diff
@@ -219,6 +224,63 @@ var _ = require('underscore'),
         $scope.search();
       };
 
+      $scope.simprintsEnabled = !!(window.medicmobile_android &&
+          window.medicmobile_android.simprints_available &&
+          window.medicmobile_android.simprints_available());
+      $scope.triggerSimprintsIdent = function() {
+        /* jslint bitwise: true */
+        var simprintsInputId = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER) & SP_ID_MASK;
+        console.log('Launching simprints with input ID: ' + simprintsInputId);
+
+        var $input = $($element).find('input[name=simprints-idents]');
+        $input.attr('data-simprints-idents', simprintsInputId);
+        $input.on('change', function() {
+          var $this = $(this);
+          $this.off('change');
+          handleSimprintsIdentResponse(JSON.parse($this.val()));
+        });
+
+        $scope.loading = true;
+        window.medicmobile_android.simprints_ident(simprintsInputId);
+      };
+
+      var handleSimprintsIdentResponse = function(idents) {
+        console.log('handleSimprintsIdentResponse()', idents);
+
+        $scope.loading = true;
+        $scope.filtered = true;
+        liveList = LiveList['contact-simprints-search'];
+        liveList.set([]);
+
+        DB().query('medic-client/people_by_simprints_id', {
+          keys: _.pluck(idents, 'id'),
+          include_docs: true,
+        })
+          .then(function(results) {
+            var docs = _.pluck(results.rows, 'doc');
+            docs.forEach(function(doc) {
+              var ident = idents.find(function(ident) {
```

Comment:
Someone is welcome to do so again, but from memory I looked into this and jshint isn't good enough to do this kind of thing. It can't tell that `idents` is an array and so there won't be the find function on it.

We could possibly just ban those function names entirely if we never think we're going to use them.

(Gosh types would be nice… ;-))